src/Microsoft.AspNetCore.Authentication/Events/AuthenticateResultContext.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    /// <summary>
+    /// Base context for authentication.
+    /// </summary>
+    public abstract class AuthenticateResultContext<TOptions> : BaseContext<TOptions> where TOptions : AuthenticationSchemeOptions
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="scheme">The authentication scheme.</param>
+        /// <param name="options">The authentication options associated with the scheme.</param>
+        protected AuthenticateResultContext(
+            HttpContext context,
+            AuthenticationScheme scheme,
+            TOptions options)
+            : base(context, scheme, options)
+        {
+        }
+
+        /// <summary>
+        /// Gets or set the <see cref="AuthenticationTicket"/> containing
+        /// the user principal and the authentication properties.
+        /// </summary>
+        public AuthenticationTicket Ticket { get; set; }
+
+        /// <summary>
+        /// Gets the <see cref="ClaimsPrincipal"/> containing the user claims.
+        /// </summary>
+        public ClaimsPrincipal Principal => Ticket?.Principal;
+
+        /// <summary>
+        /// Gets or sets the <see cref="AuthenticationProperties"/>.
+        /// </summary>
+        public override AuthenticationProperties Properties
```

Comment:
> Oh ewww, I didn't see the change to set. I'd just change the override to throw, block setting Properties in ResultContext, Properties => Ticket.Properties only.

@HaoK unfortunately, we can't make the setter throw 'cause it's used by events like `MessageReceived` where there's no ticket available (only properties).

> Should you be setting both Ticket and base,Properties rather than either/or?
 
Maybe.

I'll probably have to change the getter to return `base.Properties` first if it's explicitly set because we have OIDC events that set both the ticket and the properties with potentially different instances (e.g `TokenValidated` or `AuthorizationCodeReceived`). Or maybe I should just update the events that set the two properties to only set the ticket?